crates/bevy_render/src/mesh/allocator.rs

```diff
@@ -0,0 +1,1025 @@
+//! Manages mesh vertex and index buffers.
+
+use std::{
+    borrow::Cow,
+    fmt::{self, Display, Formatter},
+    ops::Range,
+    vec::Vec,
+};
+
+use bevy_app::{App, Plugin};
+use bevy_asset::AssetId;
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::{
+    schedule::IntoSystemConfigs as _,
+    system::{Res, ResMut, Resource},
+    world::{FromWorld, World},
+};
+use bevy_utils::{
+    hashbrown::{HashMap, HashSet},
+    tracing::error,
+};
+use offset_allocator::{Allocation, Allocator};
+use wgpu::{
+    util::BufferInitDescriptor, BufferDescriptor, BufferUsages, CommandEncoderDescriptor,
+    DownlevelFlags, COPY_BUFFER_ALIGNMENT,
+};
+
+use crate::{
+    mesh::{GpuMesh, Indices, Mesh, MeshVertexBufferLayouts},
+    render_asset::{prepare_assets, ExtractedAssets},
+    render_resource::Buffer,
+    renderer::{RenderAdapter, RenderDevice, RenderQueue},
+    Render, RenderApp, RenderSet,
+};
+
+/// A plugin that provides memory management services for mesh data.
```

Comment:
```suggestion
/// A plugin that provides GPU buffer management for uploading [`Mesh`] data.
```

Something like this, maybe. Still not quite happy with it though.

Reply:
I reworded it.